zerver/views/auth.py

```diff
@@ -716,6 +716,7 @@ def api_get_server_settings(request: HttpRequest) -> HttpResponse:
     result = dict(
         authentication_methods=get_auth_backends_data(request),
         zulip_version=ZULIP_VERSION,
+        push_notification_bouncer_enabled=settings.PUSH_NOTIFICATION_BOUNCER_URL is not None,
```

Comment:
This will take a bit more logic, as I described just now in https://github.com/zulip/zulip-mobile/issues/1507#issuecomment-352598502 . Since it'll get longer than this, let's pull it out into a function, and put the function in `zerver.lib.push_notifications`.

Also since it won't be just about the bouncer, let's call it something like `push_notifications_enabled`.

The logic should be something like
* Push notifications are enabled if either direct push notifications are enabled, or a push notification bouncer is enabled.
* The bouncer is enabled if the bouncer URL is set -- and also ZULIP_ORG_ID and ZULIP_ORG_KEY are set. Even then there could be something wrong in the config, but having these three set at all is a bare minimum.
* Direct push notifications are enabled if both GCM and APNs are configured. (I don't think it's worth adding the complexity to handle a hypothetical case where one but not the other is.)

Reply:
@gnprice Sorry for the long delay! 

I've updated the PR :)